src/go/k8s/tests/e2e-v2/upgrade-rollback/01-upgrade-bad-redpanda.yaml

```diff
@@ -7,6 +7,10 @@ spec:
   chartRef:
     timeout: 1m
     chartVersion: "5.3.2"
+    upgrade:
+      remediation:
+        retries: 1
+        strategy: "rollback"
   clusterSpec:
     image:
       tag: v23.99.99
```

Comment:
I'm not sure if `src/go/k8s/tests/e2e-v2/upgrade-rollback/01-assert-upgrade-bad-redpanda.yaml` could assert your new condition?

in https://github.com/redpanda-data/redpanda-operator/blob/c425a9ef7d5241226c638d4efc2ed822511c4cbe/src/go/k8s/tests/e2e-v2/upgrade-rollback/01-assert-upgrade-bad-redpanda.yaml#L2-L17
```
apiVersion: cluster.redpanda.com/v1alpha2
kind: Redpanda
metadata:
  finalizers:
    - operator.redpanda.com/finalizer
  name: redpanda-rollback
status:
  conditions:
    - message: HelmRelease 'redpanda/redpanda-rollback' is not ready
      reason: ArtifactFailed
      status: "False"
      type: Ready
    - message: "Not all replicas updated, available, and ready for StatefulSet(s) [redpanda-rollback (updated/available/ready/total: 1/1/1/3)]"
      reason: RedpandaPodsNotReady
      status: "False"
      type: Ready
  helmRelease: redpanda-rollback
  helmReleaseReady: false
  helmRepository: redpanda-repository
  helmRepositoryReady: true
```

P.S. 
There will be time when I learn how to interpret condition 😡 . I don't like `RedpandaPodsNotReady` reason to be of type `Ready` with that status `False`. For me it make 0 sense. 
I would like to see condition RedpandaPodsReady of type Ready with status `False` when they are not ready and status `True` when they are ready.

Reply:
Got it, I can add a test assertion for the new condition. And yeah, Kubernetes conditions are always slightly weird to me, at least the way the internal objects use them, I've always thought of things like the `Ready` status as a roll-up of a bunch of sub-statuses that are reflected in the "reason" field, so if you say:

"Why is the CRD/cluster not marked ready (i.e. Ready = False)?", then you can answer it with the Reason field, "oh, it's because the Redpanda pods are not ready."

Reply:
So, I tried really hard to add a test for this, but frankly the garbage kuttl does with merge-based patching makes this next to impossible to test for me. Specifically I'm trying to get a pod into an unschedulable state and the easiest way of doing that is to allocate way more resources than are available for it. Well, once those are set on the CRD, I can't unset them due to kuttl, so I wind up needing to modify them, which is super environment and error prone.

That left me with adding an entirely new test, with the unschedulable state as the last step in the test. Well, unfortunately, for some reason that winds up having orphaned PVCs on the system that kuttl is somehow unable to clean up and need to be deleted manually. I went ahead and just updated the text for the condition itself, but I don't think I'll be able to really write a proper end to end test for this until it's done outside of kuttl.